tgui/packages/tgui/interfaces/FishAnalyzer.tsx

```diff
@@ -0,0 +1,332 @@
+import { createSearch } from 'common/string';
+import { useState } from 'react';
+import { BooleanLike } from 'tgui-core/react';
+
+import { useBackend } from '../backend';
+import {
+  Box,
+  Button,
+  DmIcon,
+  Flex,
+  Icon,
+  Input,
+  ProgressBar,
+  Section,
+  Stack,
+} from '../components';
+import { Window } from '../layouts';
+
+type fishData = {
+  fish_name: string;
+  fish_icon: string;
+  fish_icon_state: string;
+  fish_health: number;
+  fish_size: number;
+  fish_weight: number;
+  fish_food: string;
+  fish_food_color: string;
+  fish_min_temp: number;
+  fish_max_temp: number;
+  fish_hunger: number;
+  fish_fluid_compatible: BooleanLike;
+  fish_fluid_type: string;
+  fish_suitable_temp: BooleanLike;
+  fish_breed_timer: number;
+  fish_traits: traitData[];
+  fish_evolutions: evolutionData[];
+};
+
+type traitData = {
+  trait_name: string;
+  trait_desc: string;
+  trait_inherit: number;
+};
+
+type evolutionData = {
+  evolution_name: string;
+  evolution_icon: string;
+  evolution_icon_state: string;
+  evolution_probability: number;
+  evolution_conditions: string;
+};
+
+type Data = {
+  fish_list: fishData[];
+  fish_scanned: BooleanLike;
+};
+
+export const FishAnalyzer = (props) => {
```

Comment:
this component is pretty huge, perhaps could be split into subcomponents at each `.map`

ideal components are between 30-100 lines (loose)